app/models/user.rb

```diff
@@ -63,6 +67,27 @@ def is_application?
     false
   end
 
+  # State helpers.
+  #
+  # A User model begins life in the "temp" state, and can then be claimed by another user
+  # who originated from an OAuth login. Upon it being claimed it will be removed and it's
+  # data merged with the claimant.
+  #
+  # A User can also be created by a one of the consumer applications as a stand-in
+  # for a person who has not yet (or may never) created an account.  In this case
+  # the User model will be in the "unclaimed" state.  When the User does signup, they
```

Comment:
The implementation is in https://github.com/openstax/accounts/blob/temp-users/app/routines/merge_unclaimed_users.rb  It's called from the `ContactInfosConfirm` handler https://github.com/openstax/accounts/blob/temp-users/app/handlers/contact_infos_confirm.rb#L18

Feature spec for it: https://github.com/openstax/accounts/blob/temp-users/spec/features/user_claims_account_spec.rb